crates/rolldown_testing/src/hmr-runtime.js

```diff
@@ -0,0 +1,32 @@
+// @ts-check
+
+/// <reference path="../../../crates/rolldown/src/runtime/runtime-extra-dev-common.js" />
+
+class TestDevRuntime extends DevRuntime {
+  /**
+   * @override
+   * @param {string} _moduleId
+   */
+  createModuleHotContext(_moduleId) {
+    return { accept() {} };
+  }
+  /**
+   * @override
+   * @param {string[]} _boundaries
+   */
+  applyUpdates(_boundaries) {
+    // do nothing
+  }
+}
+
+(/** @type {any} */ (globalThis)).__rolldown_runtime__ ??= new TestDevRuntime();
```

Comment:
Should we use the real runtime to test the behaviors?

Reply:
The real runtime uses WebSocket, so we need to start a WebSocket server and allocate unique ports for it and I wanted to avoid that for now.